gr-digital/lib/crc32_bb_impl.cc

```diff
@@ -65,21 +65,40 @@ namespace gr {
       }
     }
 
+    unsigned int
+    crc32_bb_impl::calculate_crc32(const unsigned char* in, size_t packet_length){
+      unsigned int crc = 0;
+      d_crc_impl.reset();
+      if (!d_packed){
+        const size_t n_packed_length = 1 + ((packet_length - 1) / 8);
+        unsigned char packed_buffer[n_packed_length];
+        memset(packed_buffer, 0, n_packed_length);
+        for (size_t bit = 0; bit < packet_length; bit++){
+          packed_buffer[bit/8] |= (in[bit] << (bit % 8));
+        }
+        d_crc_impl.process_bytes(packed_buffer, n_packed_length);
+        crc = d_crc_impl();
+      } else{
+        d_crc_impl.process_bytes(in, packet_length);
+        crc = d_crc_impl();
+      }
+      return crc;
+    }
+
     int
     crc32_bb_impl::work(int noutput_items,
                         gr_vector_int &ninput_items,
                         gr_vector_const_void_star &input_items,
                         gr_vector_void_star &output_items) {
       const unsigned char *in = (const unsigned char *) input_items[0];
       unsigned char *out = (unsigned char *) output_items[0];
-      long packet_length = ninput_items[0];
+      size_t packet_length = ninput_items[0];
       int packet_size_diff = d_check ? -d_crc_length : d_crc_length;
       unsigned int crc;
 
       if (d_check) {
-        d_crc_impl.reset();
         d_crc_impl.process_bytes(in, packet_length - d_crc_length);
-        crc = d_crc_impl();
+        crc = calculate_crc32(in, packet_length - d_crc_length);
```

Comment:
looks like if `ninput_items[0] == d_crc_length`, you call `calculate_crc32` with an input argument of packet_length = 0

Reply:
Will this ever happen for a valid crc32 packet? If yes will it explode?